tests/rptest/services/redpanda.py

```diff
@@ -37,6 +38,58 @@
 MetricSample = collections.namedtuple(
     'MetricSample', ['family', 'sample', 'node', 'value', 'labels'])
 
+DEFAULT_LOG_ALLOW_LIST = [
+    # Tests currently don't run on XFS, although in future they should.
+    # https://github.com/vectorizedio/redpanda/issues/2376
+    re.compile("not on XFS. This is a non-supported setup."),
+
+    # This is expected when tests are intentionally run on low memory configurations
+    re.compile(r"Memory: '\d+' below recommended"),
+    # A client disconnecting is not bad behaviour on redpanda's part
+    re.compile(r"kafka rpc protocol.*(Connection reset by peer|Broken pipe)")
+]
+
+# Log errors that are expected in tests that restart nodes mid-test
+RESTART_LOG_ALLOW_LIST = [
+    re.compile("(raft|rpc) - .*(disconnected_endpoint)"),
+]
+
+# Log errors that are expected in chaos-style tests that e.g.
+# stop redpanda nodes uncleanly
+CHAOS_LOG_ALLOW_LIST = [
+    # Unclean connection shutdown
+    re.compile(
+        "(raft|rpc) - .*(client_request_timeout|disconnected_endpoint|Broken pipe|Connection reset by peer)"
+    ),
+
+    # Torn disk writes
+    re.compile("storage - Could not parse header"),
+
+    # e.g. raft - [group_id:59, {kafka/test-topic-319-1639161306093460/0}] consensus.cc:2301 - unable to replicate updated configuration: raft::errc::replicated_entry_truncated
+    re.compile("raft - .*replicated_entry_truncated"),
+
+    # e.g. cluster - controller_backend.cc:466 - exception while executing partition operation: {type: update_finished, ntp: {kafka/test-topic-1944-1639161306808363/1}, offset: 413, new_assignment: { id: 1, group_id: 65, replicas: {{node_id: 3, shard: 2}, {node_id: 4, shard: 2}, {node_id: 1, shard: 0}} }, previous_assignment: {nullopt}} - std::__1::__fs::filesystem::filesystem_error (error system:39, filesystem error: remove failed: Directory not empty [/var/lib/redpanda/data/kafka/test-topic-1944-1639161306808363])
+    re.compile("cluster - .*Directory not empty"),
+    re.compile("r/heartbeat - .*cannot find consensus group:3"),
+
+    # raft - [follower: {id: {1}, revision: {9}}] [group_id:1, {kafka/topic-xyeyqcbyxi/0}] - recovery_stm.cc:422 - recovery append entries error: rpc::errc::exponential_backoff
+    re.compile("raft - .*recovery append entries error"),
+
+    # rpc - Service handler threw an exception: std::exception (std::exception)
```

Comment:
This one is kind of surprising, somewhere we're picking up a bare std::exception with no more context while a busy node shuts down in node_operations_test

Reply:
hmm, this one is fun. other than attaching a debugger, I think we're stuck working our way backwards. maybe we can start by adding the name (or the numeric id) of the method handler to the log message `rpc - Service handler threw an exception`.

there is a nice thing in Seastar which is like `Seastar::throw_with_backtrace` which perhaps we should be using more often, but I don't see any evidence of a bare std::exception being raised by redpanda :/